hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -268,10 +268,10 @@ def accept(self):
     def __init__(
         self,
         engine: "ConjectureRunner",
-        initial: ConjectureData,
-        predicate: Optional[Callable[..., bool]],
+        initial: Union[ConjectureData, ConjectureResult],
+        predicate: Optional[Callable[[ConjectureData], bool]],
         *,
-        allow_transition: bool,
+        allow_transition: Optional[bool],
```

Comment:
Ah, I think this annotation was _completely_ wrong before, and it's actually an `Optional[Callable[[Union[ConjectureData, ConjectureResult]], bool]]`.

(can't wait for py39 to go EOL, `Callable[[ConjectureData | ConjectureResult], bool] | None` is much nicer)

Reply:
The Callable should have two arguments, no? On line 289, `self.__allow_transition = allow_transition or (lambda source, destination: True)`, would suggest that it takes in two arguments. And it's usage on Line 424 (`self.__allow_transition(self.shrink_target, data)`) suggests that the second argument should be `ConjectureData`.

Reply:
`Optional[Callable[[Union[ConjectureData, ConjectureResult], ConjectureData], bool]]` seems to indeed be the right type that's consistent with usage in `shrinker.py`.